x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/container.tsx

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiFlyoutHeader, EuiSpacer, EuiTitle } from '@elastic/eui';
+import { METRIC_TYPE } from '@kbn/analytics';
+
+import { EnrichedDeprecationInfo, ReindexStatus } from '../../../../../../../common/types';
+
+import type { IndexStateContext } from '../context';
+import { DeprecationBadge } from '../../../../shared';
+import {
+  UIM_REINDEX_READONLY_CLICK,
+  UIM_REINDEX_READONLY_RETRY_CLICK,
+  UIM_REINDEX_START_CLICK,
+  UIM_REINDEX_STOP_CLICK,
+  UIM_REINDEX_UNFREEZE_CLICK,
+  UIM_REINDEX_UNFREEZE_RETRY_CLICK,
+  uiMetricService,
+} from '../../../../../lib/ui_metric';
+import {
+  ReindexDetailsFlyoutStep,
+  UnfreezeDetailsFlyoutStep,
+  UpdateIndexFlyoutStep,
+  ReindexFlyoutStep,
+  WarningFlyoutStep,
+  type FlyoutStep,
+} from './steps';
+
+export interface IndexFlyoutProps extends IndexStateContext {
+  deprecation: EnrichedDeprecationInfo;
+  closeFlyout: () => void;
+}
+
+export const IndexFlyout: React.FunctionComponent<IndexFlyoutProps> = ({
+  reindexState,
+  startReindex,
+  cancelReindex,
+  updateIndexState,
+  updateIndex,
+  closeFlyout,
+  deprecation,
+}) => {
+  const { status: reindexStatus, reindexWarnings } = reindexState;
+  const { status: updateIndexStatus } = updateIndexState;
+  const { index, correctiveAction } = deprecation;
+
+  const [flyoutStep, setFlyoutStep] = useState<FlyoutStep>('details');
+
+  useEffect(() => {
+    switch (reindexStatus) {
+      case ReindexStatus.failed:
+      case ReindexStatus.fetchFailed:
+      case ReindexStatus.cancelled:
+      case ReindexStatus.inProgress:
+      case ReindexStatus.completed: {
+        setFlyoutStep('reindexing');
+        break;
+      }
+      default: {
+        setFlyoutStep('details');
+        break;
+      }
+    }
+  }, [reindexStatus]);
+
+  // useMemo(() => reindexStatusToFlyoutStep(), [reindexStatusToFlyoutStep]);
+  // useMemo(() => updateStatusToFlyoutStep(), [updateStatusToFlyoutStep]);
```

Comment:
```suggestion
```

Reply:
Good catch, removed in https://github.com/elastic/kibana/pull/210199/commits/c6748f28845edc5bf07584efc72b73e00982f222